src/FormatNumber.elm

```diff
@@ -139,5 +142,33 @@ import FormatNumber.Stringfy exposing (stringfy)
 format : Locales.Locale -> Float -> String
 format locale number_ =
     number_
-        |> parse locale
+        |> Parser.parse locale
         |> stringfy locale
+
+
+{-| Parses a pretty string into `Maybe Float`:
+
+    import FormatNumber.Locales exposing (Decimals(..), Locale, base, frenchLocale, spanishLocale, usLocale)
+
+    parse { base | thousandSeparator = "," } "100,000.345"
+    --> Just 100000.345
+
+    parse { base | thousandSeparator = ",", negativePrefix = "-" } "-100#000"
+    --> Just -100000
```

Comment:
Why the `#` in the `-100#000` example? If `#` is not a thousand separators, this example seems wrong, right?

Reply:
Yes. It is wrong. But, this is the problem with current approach of cleaning integer part of non-digits. Thousand separator does not have any value in parsing.

Reply:
We can opt for a strategy to validate every every non-digit is one of the expected tokens (negative prefix or suffix, zero prefix or suffix, separators):

1. Split the value by the regex `\d+`
2. Check if the resulting list contains only expected tokens